res/job_queue/job_queue_node.py

```diff
@@ -154,7 +154,10 @@ def _should_be_killed(self):
 
     def _job_monitor(self, driver, pool_sema, max_submit):
 
-        self.submit(driver)
+        submit_status = self.submit(driver)
+        if submit_status != 0:
```

Comment:
This is a magic `0`. It would be preferable if we can expose the status enum in Python and use it instead...